File: src/leetcodepython/tree/tree_node.py
# -*- coding:utf-8 -*-
'''
author:zhangyu
date:2020/3/13
'''
from typing import List


class TreeNode:
    def __init__(self, x=0):
        self.val = x
        self.left = None
        self.right = None

    def create_binary_tree_array(self, array:List[int], index=0):
        '''
            将数组构造成二叉树
        Args:
            array:数组
            index:开始下标
        Returns:
            二叉树
        '''
        tree_node = None
        if index < len(array):
            value = array[index]
            if not value:
                return None
            tree_node = TreeNode(value)
            tree_node.left = self.create_binary_tree_array(array, 2 * index + 1)
            tree_node.right = self.create_binary_tree_array(array, 2 * index + 2)
            return tree_node
        return tree_node


if __name__ == '__main__':
    arr = [5, 2, 13]
    tree_node = TreeNode()
    tree = tree_node.create_binary_tree_array(arr)
    print(tree)

File: src/leetcodejava/tree/TreeNode.java
package leetcodejava.tree;

/**
 * 二叉树类
 *
 * @author: zhangyu
 */
public class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode() {
    }

    public TreeNode(int x) {
        val = x;
    }


    /**
     * 将数组构建为二叉树
     *
     * @param array 二维数组
     * @return 二叉树
     */
    public static TreeNode createBinaryTreeByArray(Integer[] array) {
        return createBinaryTreeByArray(array, 0);
    }

    /**
     * 将数组构建为二叉树
     *
     * @param array 二维数组
     * @param index 下标
     * @return 二叉树
     */
    public static TreeNode createBinaryTreeByArray(Integer[] array, Integer index) {
        if (index == null) {
            index = 0;
        }
        TreeNode treeNode = null;
        if (index < array.length) {
            Integer value = array[index];
            if (value == null) {
                return null;
            }
            treeNode = new TreeNode(value);
            treeNode.left = createBinaryTreeByArray(array, 2 * index + 1);
            treeNode.right = createBinaryTreeByArray(array, 2 * index + 2);
            return treeNode;
        }
        return treeNode;
    }
}
